packages-exp/auth-exp/src/core/user/additional_user_info.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  AdditionalUserInfo,
+  UserProfile,
+  ProviderId
+} from '@firebase/auth-types-exp';
+import { IdTokenResponse, IdTokenResponseKind } from '../../model/id_token';
+import { _parseToken } from './id_token_result';
+
+/**
+ * Parse the `AdditionalUserInfo` from the ID token response.
+ */
+export function _fromIdTokenResponse(
+  idTokenResponse: IdTokenResponse,
+): AdditionalUserInfo | null {
```

Comment:
I just raised a big PR to make UserCredential accept generics properly. Not for this PR but eventually we'll probably want to make this function generic as well.

See for example: https://github.com/firebase/firebase-js-sdk/pull/3266/files#diff-4e1f7cb1a27ae1754eb03b583f5607ccL389